stylelint-polaris/plugins/global-disallowed-list/index.js

```diff
@@ -28,6 +28,35 @@ const {rule} = stylelint.createPlugin(
         );
       }
 
+      // Mixin detection
+      postcssRoot.walkAtRules((atRule) => {
+        const value = atRule.source.input.css.slice(
+          atRule.source.start.offset,
+          atRule.source.end.offset,
+        );
```

Comment:
Thoughts on abstracting this out to a dedicated plugin e.g. `mixin-disallowed-list`?

We can then refine the experience to allow users to validate simpler string/regex matches (as right now we're working within the constraints of the `global-disallowed-list` configuration).

Take [this example](https://astexplorer.net/#/gist/407137c9c881ddf2750943bae62af165/a76a42203bbdee64805f4d6793adefa42f8c0ecf):
<img width="312" alt="image" src="https://user-images.githubusercontent.com/32409546/184196524-16667300-42ec-4689-8b57-f6609a269051.png">

<img width="527" alt="image" src="https://user-images.githubusercontent.com/32409546/184196418-e27376db-a1ee-4316-a2ac-9d1179f67265.png">

To match the above with the current implementation:
```js
/@include\s+text-style-display-large/
```
Whereas a dedicated plugin could just validate the mixin name:
```js
postcssRoot.walkAtRules('include', (atRule) => {
  const mixinName = atRule.params
  const found = primary.filter((test) => {
    return matchesStringOrRegExp(value, test);
  });
  ...
```

And is this case would be a simple string match:
```js
'stylelint-polaris/mixin-disallowed-list': ['my-mixin']
```